llama-index-integrations/retrievers/llama-index-retrievers-bm25/llama_index/retrievers/bm25/base.py

```diff
@@ -77,22 +77,18 @@ def from_defaults(
             verbose=verbose,
         )
 
-    def _get_scored_nodes(self, query: str) -> List[NodeWithScore]:
-        tokenized_query = self._tokenizer(query)
-        doc_scores = self.bm25.get_scores(tokenized_query)
-
-        nodes: List[NodeWithScore] = []
-        for i, node in enumerate(self._nodes):
-            nodes.append(NodeWithScore(node=node, score=float(doc_scores[i])))
-
-        return nodes
-
     def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
         if query_bundle.custom_embedding_strs or query_bundle.embedding:
             logger.warning("BM25Retriever does not support embeddings, skipping...")
 
-        scored_nodes = self._get_scored_nodes(query_bundle.query_str)
+        query = query_bundle.query_str
+        tokenized_query = self._tokenizer(query)
+        scores = self.bm25.get_scores(tokenized_query)
+
+        top_n = scores.argsort()[::-1][: self._similarity_top_k]
```

Comment:
seems like it would be even faster if we didn't reverse the list

`top_n = scores.argsort()[:-self._similarity_top_k]`